From 7db43407d72c399d20b94ba07a8ffb47e6506fd0 Mon Sep 17 00:00:00 2001
From: Grace Cham <hscham@chromium.org>
Date: Fri, 14 Oct 2022 14:41:31 +0900
Subject: [PATCH] disable allocator sanitizer

Disable custom memory allocator when asan is used.

Change-Id: I593b2acc1b51b8fbd310acdff264f58cd4947773
patch-name: long-term-0200-disable-allocator-sanitizer.patch
---
 .../shim/allocator_shim_override_cpp_symbols.h                | 4 ++++
 .../shim/allocator_shim_override_libc_symbols.h               | 4 ++++
 2 files changed, 8 insertions(+)

diff --git a/base/allocator/partition_allocator/src/partition_alloc/shim/allocator_shim_override_cpp_symbols.h b/base/allocator/partition_allocator/src/partition_alloc/shim/allocator_shim_override_cpp_symbols.h
index 5ec1625bf8..66e2b98774 100644
--- a/base/allocator/partition_allocator/src/partition_alloc/shim/allocator_shim_override_cpp_symbols.h
+++ b/base/allocator/partition_allocator/src/partition_alloc/shim/allocator_shim_override_cpp_symbols.h
@@ -31,6 +31,9 @@
 #define SHIM_CPP_SYMBOLS_EXPORT PA_NOINLINE
 #endif
 
+// Disable custom memory allocator when asan is used.
+// https://crbug.com/807685
+#if !defined(DISABLE_ALLOCATOR_SANITIZER)
 SHIM_CPP_SYMBOLS_EXPORT void* operator new(size_t size) {
   return ShimCppNew(size);
 }
@@ -130,5 +133,6 @@ SHIM_CPP_SYMBOLS_EXPORT void operator delete[](void* p,
                                                const std::nothrow_t&) __THROW {
   ShimCppDelete(p);
 }
+#endif
 
 #endif  // BASE_ALLOCATOR_PARTITION_ALLOCATOR_SRC_PARTITION_ALLOC_SHIM_ALLOCATOR_SHIM_OVERRIDE_CPP_SYMBOLS_H_
diff --git a/base/allocator/partition_allocator/src/partition_alloc/shim/allocator_shim_override_libc_symbols.h b/base/allocator/partition_allocator/src/partition_alloc/shim/allocator_shim_override_libc_symbols.h
index 624a44610e..05030a8b1c 100644
--- a/base/allocator/partition_allocator/src/partition_alloc/shim/allocator_shim_override_libc_symbols.h
+++ b/base/allocator/partition_allocator/src/partition_alloc/shim/allocator_shim_override_libc_symbols.h
@@ -22,6 +22,9 @@
 
 #include "base/allocator/partition_allocator/src/partition_alloc/shim/allocator_shim_internals.h"
 
+// Disable custom memory allocator when asan is used.
+// https://crbug.com/807685
+#if !defined(DISABLE_ALLOCATOR_SANITIZER)
 extern "C" {
 
 // WARNING: Whenever a new function is added there (which, surprisingly enough,
@@ -88,5 +91,6 @@ SHIM_ALWAYS_EXPORT size_t malloc_usable_size(void* address) __THROW {
 //   struct mallinfo mallinfo(void);
 
 }  // extern "C"
+#endif
 
 #endif  // BASE_ALLOCATOR_PARTITION_ALLOCATOR_SRC_PARTITION_ALLOC_SHIM_ALLOCATOR_SHIM_OVERRIDE_LIBC_SYMBOLS_H_
-- 
2.42.0.582.g8ccd20d70d-goog

